### PR TITLE
Changed video processing section (Unit 7 - CNN Based Video Model)

### PR DESCRIPTION
**Pull request by cjfghk5697**
Changed`chapters/en/unit7/video-processing/overview-of-previous-sota-models.mdx` to `chapters/en/unit7/video-processing/cnn-based-video-model.mdx`  at video processing section. This document provides an overview of various CNN video architectures which integrate different kinds of modalities into a unified representation space.

Part of https://github.com/johko/computer-vision-course/discussions/348

Who can review? (Initial)
@jungnerd @cjfghk5697 @mreraser and anyone who wants to review!

### CODE REVIEW

**Comment by mreraser** [comment]
Hello @cjfghk5697 😄  I think you haven't modified _toctree.yml file yet. 

```
RuntimeError: The following files are not present in the table of contents:
- unit7/video-processing/cnn-based-video-model
Add them to ../computer-vision-course/chapters/en/_toctree.yml.
```

File: chapters/en/_toctree.yml [hunk 1]
```diff
@@ -124,8 +124,8 @@
     local: "unit7/video-processing/introduction-to-video"
   - title: Video Processing Basics
     local: "unit7/video-processing/video-processing-basics"
-  - title: Overview of the previous SOTA models
-    local: "unit7/video-processing/overview-of-previous-sota-models"
+  - title: Cnn Based Video Model
```
Comment (johko) [inline chapters/en/_toctree.yml:127]:
```suggestion
  - title: CNN Based Video Model
```

File: chapters/en/unit7/video-processing/cnn-based-video-model.mdx [hunk 2]
```diff
@@ -0,0 +1,134 @@
+# CNN Based Video Model
```
Comment (johko) [inline chapters/en/unit7/video-processing/cnn-based-video-model.mdx:1]:
```suggestion
# CNN Based Video Models
```

File: chapters/en/unit7/video-processing/cnn-based-video-model.mdx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+### General Trend:
+
+The success of Deep Learning, particularly CNNs trained on massive datasets like ImageNet, revolutionized image recognition. This trend continues in video processing. However, video data introduces another dimension compared to static images: time. This simple change introduced a new set of challenges that CNN trained in static images were not built to deal with.
```
Comment (johko) [inline chapters/en/unit7/video-processing/cnn-based-video-model.mdx:5]:
```suggestion
The success of Deep Learning, particularly CNNs trained on massive datasets like ImageNet, revolutionized image recognition. This trend continues in video processing. However, video data introduces another dimension compared to static images: time. This simple change introduced a new set of challenges that CNNs trained in static images were not built to deal with.
```

File: chapters/en/unit7/video-processing/cnn-based-video-model.mdx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+# Previous SOTA Models in Video Processing
+
+## Two-Stream Network(2014):
```
Comment (johko) [inline chapters/en/unit7/video-processing/cnn-based-video-model.mdx:9]: Mainly for consistency within the course, I would remove the colon in (sub)headlines
```suggestion
## Two-Stream Network(2014)
```

File: chapters/en/unit7/video-processing/cnn-based-video-model.mdx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+This paper extended Deep Convolutional Networks(ConvNets) to perform action-recognition in video data.
+
+The proposed architecture is called Two-Stream Network. It uses two separate pathways within a neural network:
+
+- **Spatial Stream:** A standard 2D CNN processes individual frames to capture appearance information.
+- **Temporal Stream:** A 2D CNN, or another network, that processes several frame sequences (optical flow) to capture motion information.
+- **Fusion:** The outputs from both streams are then combined to leverage both appearance and motion cues for tasks like action recognition.
+
+## 3D ResNets(2017):
```
Comment (johko) [inline chapters/en/unit7/video-processing/cnn-based-video-model.mdx:23]:
```suggestion
## 3D ResNets(2017)
```

File: chapters/en/unit7/video-processing/cnn-based-video-model.mdx [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+<div class="flex justify-center">
+<img src="https://huggingface.co/datasets/hf-vision/course-assets/resolve/main/previous sota models/SOTA Models Residual block. Shortcut connections bypass a signal from the top of the block to the tail. Signals are summed at the tail..png" alt="Residual block. Shortcut connections bypass a signal from the top of the block to the tail. Signals are summed at the tail.">
+</div>
+
+Standard 3D CNNs extend the concept to simultaneously capture spatial and temporal information using 3D kernels (2D spatial information + temporal information). A drawback of this model is that the large number of parameters result the training being more computationally intensive and hence slower than the 2D version. Therefore, the 3D version of the ConvNets typically has fewer layers than the deeper architectures of 2D CNNs.
```
Comment (johko) [inline chapters/en/unit7/video-processing/cnn-based-video-model.mdx:29]:
```suggestion
Standard 3D CNNs extend the concept to simultaneously capture spatial and temporal information using 3D kernels (2D spatial information + temporal information). A drawback of this model is that the large number of parameters result in the training being more computationally intensive and hence slower than the 2D version. Therefore, the 3D version of the ConvNets typically has fewer layers than the deeper architectures of 2D CNNs.
```

File: chapters/en/unit7/video-processing/cnn-based-video-model.mdx [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+
+In this paper, the authors applied the ResNet architecture to the 3D CNNs. This approach introduces deeper models for 3D CNNs and achieves higher accuracy.
+
+Experiments showed that the 3D ResNets (especially deeper ones like the ResNet-34) outperform models like the C3D, particularly on larger datasets. Pretrained models like Sports-1M C3D can help mitigate overfitting on smaller datasets. Overall, 3D ResNets effectively leverage deeper architectures to capture complex spatiotemporal patterns in the video data.
```
Comment (johko) [inline chapters/en/unit7/video-processing/cnn-based-video-model.mdx:33]: Here you mention `C3D` as if it is already known, maybe shortly introduce it or give a link to the paper?

File: chapters/en/unit7/video-processing/cnn-based-video-model.mdx [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+
+| Method | Validation set |  |  | Testing set |  |  |
+| --- | --- | --- | --- | --- | --- | --- |
+|  | Top-1 | Top-5 | Average | Top-1 | Top-5 | Average |
+| 3D ResNet-34 | 58.0 | 81.3 | **69.7** | - | - | **68.9** |
+| C3D* | 55.6 | 79.1 | 67.4 | 56.1 | 79.5 | 67.8 |
+| C3D w/ BN | 56.1 | 79.5 | 67.8 | - | - | - |
+| RGB-I3D w/o ImageNet | - | - | 68.4 | 88.0 | **78.2** |  |
+
+## (2+1)D ResNets(2017):
```
Comment (johko) [inline chapters/en/unit7/video-processing/cnn-based-video-model.mdx:43]:
```suggestion
## (2+1)D ResNets(2017)
```

File: chapters/en/unit7/video-processing/cnn-based-video-model.mdx [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+
+**Key Features**
+
+- **Graph-Based Modeling**: Represents human skeletal data as graphs, allowing for natural modeling of joint connections.
+- **Spatio-Temporal Convolutions**: Integrates spatial and temporal graph convolutions to capture dynamic movement patterns.
+- **Real-Time Performance**: Optimized for fast computation, making it suitable for real-time applications.
+
+# Conclusion
+
+The evolution of video analysis models has been fascinating to witness. These models were heavily influenced by other SOTA models. For example, Two-StreamNets was motivated by the  ConvNets and (2+1)D ResNets were inspired by the 3D ResNets. As the research progresses, one can expect even more advanced architectures and techniques to emerge in the future.
```
Comment (johko) [inline chapters/en/unit7/video-processing/cnn-based-video-model.mdx:134]:
```suggestion
The evolution of video analysis models has been fascinating to witness. These models were heavily influenced by other SOTA models. For example, Two-StreamNets was motivated by the ConvNets and (2+1)D ResNets were inspired by the 3D ResNets. As the research progresses, one can expect even more advanced architectures and techniques to emerge in the future.
```

**Review by johko — CHANGES_REQUESTED** [review]
Thank you for the contribution, was a pleasure to read and learn :slightly_smiling_face: 

I just left some smaller suggestions, mainly formatting related

File: chapters/en/unit7/video-processing/cnn-based-video-model.mdx (hunk 7 above)
Comment (cjfghk5697) [reply chapters/en/unit7/video-processing/cnn-based-video-model.mdx]: Thank you for reviewing it! I’ve added C3D research paper's link.

**Review by ATaylorAerospace — APPROVED** [review]
LGTM

**Review by johko — APPROVED** [review]
Thanks for the fixes. LGTM :+1:

**Pull request merged**